### ISSUE TITLE
Tabline disappears when closing a Floaterm terminal

### ISSUE THREAD

**Issue by noib3** [pacha/vem-tabline#19]
When using the [vim floaterm](https://github.com/voldikss/vim-floaterm) plugin, after closing a floating terminal (to test this set `g:floaterm_autoclose` to 2, then call `:FloatermNew sh`, then `exit` the shell), the tabline disappears.

After a few seconds it reappears automatically, but it's still annoying that it does that. I've tested other tabline plugins and they don't seem to have this particular problem.

Can this be fixed?

Edit: in general the tabline doesn't disapper, however the "tab" with the current buffer does. The tabline disappears if there are only two buffers one, since when one if them disappears the whole tabline is hidden.

**Comment by pacha** [comment]
Hi n0ibe. Not a user of vim floaterm, but just did try what you describe and it does look annoying. It seems that closing the floaterm popup triggers a `BufDelete` event with the wrong buffer number. Fortunately, the `BufUnload` event has the correct buffer number in this case. Since `BufUnload` is arguably more correct than `BufDelete` anyways I updated the code accordingly in bde613a4be3a3e9a2b51516024ff5d49ae474f05.

So, long story short, it should be fixed in master now :) 
Let me know if it works for you or if you find any problem with it.

**Comment by noib3** [comment]
@pacha Works without a problem!

**Comment by pacha** [comment]
@n0ibe Good to hear :grin:
